controller/appcontroller.go

```diff
@@ -567,17 +568,18 @@ func setApplicationHealth(comparisonResult *appv1.ComparisonResult) (*appv1.Heal
 				return nil, err
 			}
 			healthState, err := health.GetAppHealth(&obj)
-			if err != nil {
-				return nil, err
-			}
+			poErr = err
```

Comment:
I think we only need to keep track the first error, as opposed to poErr and savedErr. Basically if *any* error occurs, we will set it once (and not override it). The first error that occurs (if any), will be the one we return as a result of this function.